### PR TITLE
Fix prototype keys causing crash in color parsing and invalid colors in color coercion expressions

### PR DESCRIPTION
**Pull request by cxcorp**
From #1025

This PR fixes two issues:
1. When parsing color strings (e.g. `text-color: "red"`), `parse_css_color.ts` attempted to look up named colors from an object literal by doing `namedColors[input]`. If the color string was `"__proto__"` or `"toString"` or some other key in the `Object` prototype, the value was truthy, and caused an unhandled error to be thrown because the value was destructured.
   https://github.com/maplibre/maplibre-style-spec/blob/91175b30b548f27527ec9d3ac09b73fed31cf699/src/expression/types/parse_css_color.ts#L40-L44
2. When a `Coercion` expression was used to coerce an expression's string value to a color (e.g. `to_color`), it used `EvaluationContext.parseColor()` which checked if the string had already been parsed and cached. The cache lookup was done via `this._parseColorCache[input]`, where `input` comes from the user controlled style. As a result, the input was never parsed as the value was truthy, and the value from the prototype was returned and became the expression's value.
   https://github.com/maplibre/maplibre-style-spec/blob/91175b30b548f27527ec9d3ac09b73fed31cf699/src/expression/evaluation_context.ts#L16
   https://github.com/maplibre/maplibre-style-spec/blob/91175b30b548f27527ec9d3ac09b73fed31cf699/src/expression/evaluation_context.ts#L48-L53

Fixing issue 1 revealed issue 2, as `EvaluationContext.parseColor()` used `Color.parse()` which used `parseCssColor()`. `Color.parse()` is also used in `function` expressions.

The fixes for the issues were:
1. For `parseCssColor()`, since we probably want the convenience of using an object literal to initialize it, I opted to use the previously introduced (#1028) `getOwn()` utility to ensure that only the literal's keys are tested.
2. For `EvaluationContext.parseColor()`, the ` _parseColorCache` was previously initialized to an empty object literal `{}`. Since there was no need for the ergonomics of using a literal, I replaced it with `Object.create(null)` which creates an object without a prototype (note: `{}` is syntactic sugar for `Object.create(Object.prototype)`).

The ideal solution for this would be to systematically use a `Map` if the key is user-controlled, but I did not find any uses of it from the codebase and assumed that the library targets ES5 compatibility.

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality.
 - [x] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

File: src/expression/evaluation_context.ts [hunk 1]
```diff
@@ -20,7 +20,9 @@ export class EvaluationContext {
         this.feature = null;
         this.featureState = null;
         this.formattedSection = null;
-        this._parseColorCache = {};
+        // the cache keys are user controlled (from the source JSON), so
+        // avoid prototype pollution by creating a record with a null prototype
+        this._parseColorCache = Object.create(null) as {[_: string]: Color};
```
Comment (cxcorp) [inline src/expression/evaluation_context.ts:25]: The current TypeScript configs didn't seem to care whether this `any` was casted or not, but I added it so that it doesn't need to be fixed in the future if the TS configs are ever made stricter in the future.

File: test/integration/style-spec/tests/bad-color.output.json [hunk 2]
```diff
@@ -10,5 +10,17 @@
   {
```
Comment (cxcorp) [inline test/integration/style-spec/tests/bad-color.output.json:10]: I noticed that there were both `*.output-api-supported.json` and `*.output.json` files in the tests, but was unable to find references to `output-api-supported` or `api-supported` in the codebase, other than a changelog entry saying `--mapbox-api-supported` had been removed from `gl-style-validation` in 14.0.0. Are the `.output-api-supported.json` files still used, or subject for removal in a later PR?

File: src/expression/evaluation_context.ts (hunk 1 above)
Comment (HarelM) [reply src/expression/evaluation_context.ts:25]: If you think a Map object would be a better fit here (looks like it, since we are talking about cache) please change the code accordingly.

File: test/integration/style-spec/tests/bad-color.output.json (hunk 2 above)
Comment (HarelM) [reply test/integration/style-spec/tests/bad-color.output.json:10]: I can't find anything related to this in the history of the repo, I'm not sure they were ever used, and also from a very shallow test I made these looks duplicates of the output files, so I would recommend removing them.

File: test/integration/style-spec/tests/bad-color.output.json [hunk 3]
```diff
@@ -10,5 +10,17 @@
...
     "message": "layers[1].paint.fill-outline-color: color expected, array found",
     "line": 26
+  },
+  {
+    "message": "layers[2].paint.fill-color: color expected, \"__proto__\" found",
```
Comment (HarelM) [inline test/integration/style-spec/tests/bad-color.output.json:15]: Should the message beyter describe the problem? Is there a word missing after the "found"?

Comment (HarelM) [reply test/integration/style-spec/tests/bad-color.output.json:15]: Ignore this, my mistake...

File: src/expression/evaluation_context.ts (hunk 1 above)
Comment (cxcorp) [reply src/expression/evaluation_context.ts]: Sure, I think it's the best choice here, but was hesitant to introduce it as the first use in this repo. I thought it might be a ES5 compatibility thing, but on further investigation the codebase is already using ES6 features like `Set` and `Number.isNaN` so it's probably just a matter of this being a long-lived codebase.

Changed to use a `Map` in c52614560a3954ab939200ee98d6e309451b902d.

File: test/integration/style-spec/tests/bad-color.output.json (hunk 2 above)
Comment (cxcorp) [reply test/integration/style-spec/tests/bad-color.output.json:10]: Sure. Remove in this PR or shall I make a separate PR?

Comment (HarelM) [reply test/integration/style-spec/tests/bad-color.output.json:10]: A separate PR seems like the right approach.
